File: backend/models/inquiry.py
from pydantic import BaseModel
from typing import List, Optional


class InquiryMenuOption(BaseModel):
    id: str
    label: str
    description: str
    route: str


class InquiryMenuResponse(BaseModel):
    title: str
    options: List[InquiryMenuOption]


class PositionDetail(BaseModel):
    portfolioId: str
    investmentId: str
    positionDate: Optional[str]
    quantity: float
    costBasis: float
    marketValue: float
    currency: str
    status: str
    gainLoss: float
    gainLossPercent: float
    lastMaintDate: Optional[str]
    lastMaintUser: Optional[str]


class PortfolioInquiryResponse(BaseModel):
    accountNumber: str
    portfolioId: str
    portfolioName: Optional[str]
    clientName: Optional[str]
    clientType: Optional[str]
    status: Optional[str]
    totalValue: float
    totalCostBasis: float
    totalGainLoss: float
    totalGainLossPercent: float
    cashBalance: float
    currency: str
    positions: List[PositionDetail]
    lastUpdated: str


class HistoryEntry(BaseModel):
    transactionDate: Optional[str]
    transactionType: str
    quantity: float
    price: float
    amount: float
    investmentId: Optional[str]
    status: Optional[str]


class TransactionHistoryResponse(BaseModel):
    accountNumber: str
    portfolioId: str
    entries: List[HistoryEntry]
    totalEntries: int
    hasMore: bool
    message: str


class InquiryErrorResponse(BaseModel):
    errorCode: int
    errorMessage: str
    program: str

File: src/types/inquiry.ts
import { z } from 'zod';

export const inquiryAccountSchema = z
  .string()
  .length(10, 'Account number must be exactly 10 digits')
  .regex(/^\d+$/, 'Account number must contain only numeric characters');

export const inquiryFormSchema = z.object({
  accountNumber: inquiryAccountSchema,
});

export type InquiryFormData = z.infer<typeof inquiryFormSchema>;

export type InquiryTab = 'portfolio' | 'history';

export interface PositionDetail {
  portfolioId: string;
  investmentId: string;
  positionDate: string | null;
  quantity: number;
  costBasis: number;
  marketValue: number;
  currency: string;
  status: string;
  gainLoss: number;
  gainLossPercent: number;
  lastMaintDate: string | null;
  lastMaintUser: string | null;
}

export interface PortfolioInquiryResult {
  accountNumber: string;
  portfolioId: string;
  portfolioName: string | null;
  clientName: string | null;
  clientType: string | null;
  status: string | null;
  totalValue: number;
  totalCostBasis: number;
  totalGainLoss: number;
  totalGainLossPercent: number;
  cashBalance: number;
  currency: string;
  positions: PositionDetail[];
  lastUpdated: string;
}

export interface HistoryEntry {
  transactionDate: string | null;
  transactionType: string;
  quantity: number;
  price: number;
  amount: number;
  investmentId: string | null;
  status: string | null;
}

export interface TransactionHistoryResult {
  accountNumber: string;
  portfolioId: string;
  entries: HistoryEntry[];
  totalEntries: number;
  hasMore: boolean;
  message: string;
}
